### ISSUE TITLE
Install Command Line Tools on MacOS 11.0

### ISSUE THREAD

**Issue by maxim-lobanov** [Homebrew/install#325]
## Problem
We have an automated process of building MacOS VMs using packer. The first step of building image is installation of Homebrew. It works perfect on MacOS HS, Mojave, Catalina but fails on Big Sur (11.0).
The reason of failure is that Command Line Tools are not found.

Homebrew installs Command Line Tools automatically on HS, Mojave, Catalina but there is a condition that prevents it on 11.0: https://github.com/Homebrew/install/blob/master/install.sh#L193
Looks like this condition was added intentionally so we would like to clarify if it makes sense to remove it.

We have validated that Command Line Tools are installed perfectly using code included to brew install script: https://github.com/Homebrew/install/blob/master/install.sh#L530 (We just copy-pasted this code and run it before installing brew)

## What you were trying to do (and why)

Install Homebrew on clean MacOS VM where Command Line Tools are not installed.

## What happened (include command output)

<details>
  <summary>Command output</summary>
  <pre>

  Downloading and installing Homebrew...
  xcode-select: error: no developer tools were found at '/Applications/Xcode.app', and no install could be requested (perhaps no UI is present), please install manually from 'developer.apple.com'.

  </pre>
</details>

## What you expected to happen

Homebrew installation script installs Command Line Tools on MacOS 11.0 as on on MacOS 10.15.

**Comment by MikeMcQuaid** [comment]
We don't support 11.0 yet so don't file issues with it, please. The CLT were removed from installation intentionally (CC @Bo98).

**Comment by Bo98** [comment]
Command Line Tools installation was removed (on 11.0 only) because Apple were regularly having issues offering the latest version through `xcode-select --install`. At the time of removal, it was offering an older version on x86_64 (which was very bad because it was 10.16 while the OS was 11.0) and it was downloading a version on Apple Silicon that only worked on Intel (again, very bad and unusable). The _only_ solution was to download from developer.apple.com (or use Xcode.app).

However, this removal is only intended to be temporary. I recognise the importance this is for automation purposes. If someone can confirm that `softwareupdate`/`xcode-select` is now able to correctly install the latest version (beta 4) then we will accept a pull request to remove the 11.0 restriction.

**Comment by maxim-lobanov** [comment]
@Bo98 ,

> which was very bad because it had the 10.16 SDK while the OS was 11.0, which breaks many things

We have noticed the same. Actually, it was fixed only with Xcode 12.0 beta 3: https://github.com/actions/virtual-environments/commit/632722fedf81e05feae8349601bfbfc436568d0f#diff-e9f9822df5b3b9087dd94d4c3c083277R222

I can confirm that headless installation ([via software update](https://github.com/Homebrew/install/blob/master/install.sh#L536)) works as expected (I have tested it only on Intel):
![image](https://user-images.githubusercontent.com/16715858/90387801-7ff61400-e08f-11ea-9ff5-2f6f6a5c8ccf.png)

But [xcode-select](https://github.com/Homebrew/install/blob/master/install.sh#L553) installation fails for some reason: 
![image](https://user-images.githubusercontent.com/16715858/90387728-5ccb6480-e08f-11ea-9090-c2dacbac2d4b.png)

**Comment by Bo98** [comment]
> Actually, it was fixed only with Xcode 12.0 beta 3

Indeed. It was intentional for beta 1 and 2, and it did work for what it was designed for because beta 1 and 2 of the OS itself also reported 10.16. Beta 3 of the OS however moved away from 10.16 to 11.0, but `softwareupdate` at the time still installed beta 2 of the Command Line Tools - which is what broke things and made Homebrew unusable.

> I can confirm that headless installation (via software update) works as expected (I have tested it only on Intel):

Fantastic - thanks for trying it out! I've just confirmed with someone that it works on Apple Silicon too now.

> But xcode-select installation fails for some reason:

That's disappointing, but not the first time Apple have broken it. However, if the `softwareupdate` method works then the `xcode-select` install method should not be needed.

I've opened #327 to restore the Command Line Tools install on Big Sur.
